homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,306 @@
+            return {"base": "unable_still_load"}, None, None
+
+        if not image:
+            return {"base": "unable_still_load"}, None, None
+        fmt = imghdr.what(None, h=image)
+        if fmt is None:
+            # if imghdr can't figure it out, could be svg.
+            if image.decode("utf-8").startswith("<svg"):
+                fmt = "svg+xml"
+        _LOGGER.debug(
+            "Still image at '%s' detected format: %s",
+            info[CONF_STILL_IMAGE_URL],
+            fmt,
+        )
+        if fmt not in SUPPORTED_IMAGE_TYPES:
+            return {"base": "invalid_still_image"}, None, None
+        fmt = "image/" + fmt
+
+    # Second level functionality is to get a stream.
+    if stream_source := info.get(CONF_STREAM_SOURCE):
+        try:
+            # For RTSP streams, prefer TCP. This code is duplicated from
+            # homeassistant.components.stream.__init__.py:create_stream()
+            # It may be possible & better to call create_stream() directly.
+            stream_options: dict[str, str] = {}
+            if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
+                stream_options = {
+                    "rtsp_flags": "prefer_tcp",
+                    "stimeout": "5000000",
+                }
+            if rtsp_transport := info.get(CONF_RTSP_TRANSPORT):
+                stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = rtsp_transport
+            _LOGGER.debug("Attempting to open stream %s", stream_source)
+            container = await hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+        # pylint: disable=c-extension-no-member
+        except av.error.FileNotFoundError:
+            return {CONF_STREAM_SOURCE: "stream_no_route_to_host"}, None, None
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return {CONF_STREAM_SOURCE: "stream_unauthorised"}, None, None
+        except (KeyError, IndexError):
+            return {CONF_STREAM_SOURCE: "stream_novideo"}, None, None
+        except PermissionError:
+            return {CONF_STREAM_SOURCE: "stream_not_permitted"}, None, None
+        except OSError as err:
+            if err.errno == EHOSTUNREACH:
+                return {CONF_STREAM_SOURCE: "stream_no_route_to_host"}, None, None
+            if err.errno == EIO:  # input/output error
+                return {CONF_STREAM_SOURCE: "stream_io_error"}, None, None
+            raise err
+    name = info.get(CONF_NAME, url or stream_source)
+    return errors, fmt, name
+
+
+class GenericIPCamConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for generic IP camera."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
```

Comment:
You can drop this as it goes in the manifest now

Reply:
Thanks! 009df05c